### ISSUE TITLE
Gedmo Extension Tree not longer working with DoctrineBundle 2.3 Version

### ISSUE THREAD

**Issue by alexander-schranz** [doctrine-extensions/DoctrineExtensions#2210]
As seen in https://github.com/doctrine/DoctrineBundle/issues/1305 there where some breaking changes which will not be changed there. So we will need to adjust the gedmo extension to use the new `getDriver` which will be introduce in https://github.com/doctrine/DoctrineBundle/pull/1304. I did currently not yet get it to work but maybe @stof know more what we need to change here to get it again using gedmo over the doctrine bundle with https://github.com/stof/StofDoctrineExtensionsBundle. 

## Here the Original Issue

I updated in our project only `doctrine/doctrine-bundle` from `2.2.3` to `2.3.0`.

After the update I'm getting the following when the `cache:clear` is called:

> In TreeListener.php line 71:
>  Tree object class: Sulu\Bundle\MediaBundle\Entity\Collection must have tree metadata at this point

<details>
<summary>Stack Trace</summary>

```bash
Exception trace:
  at /private/tmp/sulu-develop.localhost/vendor/gedmo/doctrine-extensions/src/Tree/TreeListener.php:71
 Gedmo\Tree\TreeListener->getStrategy() at /private/tmp/sulu-develop.localhost/vendor/gedmo/doctrine-extensions/src/Tree/Entity/Repository/NestedTreeRepository.php:891
 Gedmo\Tree\Entity\Repository\NestedTreeRepository->validate() at /private/tmp/sulu-develop.localhost/vendor/gedmo/doctrine-extensions/src/Tree/Entity/Repository/AbstractTreeRepository.php:53
 Gedmo\Tree\Entity\Repository\AbstractTreeRepository->__construct() at /private/tmp/sulu-develop.localhost/vendor/doctrine/doctrine-bundle/Repository/ContainerRepositoryFactory.php:79
 Doctrine\Bundle\DoctrineBundle\Repository\ContainerRepositoryFactory->getOrCreateRepository() at /private/tmp/sulu-develop.localhost/vendor/doctrine/doctrine-bundle/Repository/ContainerRepositoryFactory.php:65
 Doctrine\Bundle\DoctrineBundle\Repository\ContainerRepositoryFactory->getRepository() at /private/tmp/sulu-develop.localhost/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php:758
 Doctrine\ORM\EntityManager->getRepository() at /private/tmp/sulu-develop.localhost/var/cache/website/prod/ContainerVYSWbWN/EntityManager_9a5be93.php:153
 ContainerVYSWbWN\EntityManager_9a5be93->getRepository() at /private/tmp/sulu-develop.localhost/var/cache/website/prod/ContainerVYSWbWN/App_KernelProdContainer.php:2196
 ContainerVYSWbWN\App_KernelProdContainer->getSuluMedia_CollectionRepositoryService() at /private/tmp/sulu-develop.localhost/var/cache/website/prod/ContainerVYSWbWN/App_KernelProdContainer.php:2314
 ContainerVYSWbWN\App_KernelProdContainer->getSuluMedia_MediaManagerService() at /private/tmp/sulu-develop.localhost/var/cache/website/prod/ContainerVYSWbWN/App_KernelProdContainer.php:958
 ContainerVYSWbWN\App_KernelProdContainer->get_Container_Private_TwigService() at /private/tmp/sulu-develop.localhost/vendor/symfony/dependency-injection/Container.php:441
 Symfony\Component\DependencyInjection\Container->getService() at /private/tmp/sulu-develop.localhost/vendor/symfony/dependency-injection/Argument/ServiceLocator.php:42
 Symfony\Component\DependencyInjection\Argument\ServiceLocator->get() at /private/tmp/sulu-develop.localhost/vendor/symfony/twig-bundle/CacheWarmer/TemplateCacheWarmer.php:46
 Symfony\Bundle\TwigBundle\CacheWarmer\TemplateCacheWarmer->warmUp() at /private/tmp/sulu-develop.localhost/vendor/symfony/http-kernel/CacheWarmer/CacheWarmerAggregate.php:98
 Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerAggregate->warmUp() at /private/tmp/sulu-develop.localhost/vendor/symfony/framework-bundle/Command/CacheClearCommand.php:243
 Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->warmup() at /private/tmp/sulu-develop.localhost/vendor/symfony/framework-bundle/Command/CacheClearCommand.php:155
 Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->execute() at /private/tmp/sulu-develop.localhost/vendor/symfony/console/Command/Command.php:256
 Symfony\Component\Console\Command\Command->run() at /private/tmp/sulu-develop.localhost/vendor/symfony/console/Application.php:989
 Symfony\Component\Console\Application->doRunCommand() at /private/tmp/sulu-develop.localhost/vendor/symfony/framework-bundle/Console/Application.php:96
 Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at /private/tmp/sulu-develop.localhost/vendor/symfony/console/Application.php:290
 Symfony\Component\Console\Application->doRun() at /private/tmp/sulu-develop.localhost/vendor/symfony/framework-bundle/Console/Application.php:82
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /private/tmp/sulu-develop.localhost/vendor/symfony/console/Application.php:166
 Symfony\Component\Console\Application->run() at /private/tmp/sulu-develop.localhost/bin/console.php:51
 include() at /private/tmp/sulu-develop.localhost/bin/websiteconsole:7

```

</details>

I also tried the https://github.com/doctrine/DoctrineBundle/pull/1304 PR but seems like this does not fix it, still the same error.

Not sure yet which cause the issue, happy for every hint.

PS: After debugging it in more details with git bisect it has todo with the changes in https://github.com/doctrine/DoctrineBundle/pull/1284.

#### Reproduceable Repository

```bash
composer create-project sulu/skeleton
```

```bash
bin/websiteconsole cache:clear
```

**Comment by chapterjason** [comment]
Same for Timestampable, lock the version in composer to `2.2.*` works as "workaround".

**Comment by arjanfrans** [comment]
I am having a similar issue. Since upgrading to doctrine 2.3 I get a constraint violation. I switched from annotations to XML. With annotations it seemed to work fine, but not with XML.

```
SQLSTATE[23000]: Integrity constraint violation: 19 NOT NULL constraint failed: categories.lft
```

**Comment by kbyjoel** [comment]
> Same for Timestampable, lock the version in composer to `2.2.*` works as "workaround".
> 
> I've a not null constraint for a field named `created_at` and I will get something like this with `2.3.*` `SQLSTATE[23502]: Not null violation: 7 ERROR: null value in column "created_at" of relation "user" violates not-null constraint DETAIL: Failing row contains ...`
> 
> // **EDIT**
> I debugged a bit, If I remove the metadataDriver decoration from the `IdGeneratorPass` it works.
> But I currently don't know what causes the decorator to fail the doctrine-extension.
> 
> [doctrine/DoctrineBundle@a1268d4#diff-7481aeb0edda6698c2b93b9de0475a7e6744cdbef1d0204e0584ad85e7a49b16R67-R72](https://github.com/doctrine/DoctrineBundle/commit/a1268d43a0e0e469f76bc9c027db33b23050c837#diff-7481aeb0edda6698c2b93b9de0475a7e6744cdbef1d0204e0584ad85e7a49b16R67-R72)
> 
> // **EDIT**
> I've found the issue, because the driver has been decorated, this if condition no longer applies.
> 
> https://github.com/doctrine-extensions/DoctrineExtensions/blob/7c0d5aeab0f840d2a18a18c3dc10b0117c597a42/src/Mapping/ExtensionMetadataFactory.php#L143
> 
> `$omDriver`:
> Before: `\Doctrine\Persistence\Mapping\Driver\MappingDriverChain`
> After: `\Doctrine\Bundle\DoctrineBundle\Mapping\MappingDriver`


Same observation for me ! Couldn't make Timestamble work with  `^2.3`, i had to downgrade to  `2.2.*`.
`$ormDriver` is now `Doctrine\Bundle\DoctrineBundle\Mapping\MappingDriver` instance, which has a private `driver` property of `Doctrine\Persistence\Mapping\Driver\MappingDriverChain`

**Comment by chapterjason** [comment]
For further information check here: https://github.com/doctrine/DoctrineBundle/pull/1304

They are currently looking for a better way, which doesn't break that much.

**Comment by alexander-schranz** [comment]
@AkenRoberts maybe you want to hook into there discussion what would be the best solution for the gedmo extension to get the driver https://github.com/doctrine/DoctrineBundle/pull/1304#issuecomment-806599428

**Comment by AkenRoberts** [comment]
Thank you all for the heads up. Taking a look today.

**Comment by AkenRoberts** [comment]
@alexander-schranz @chapterjason Would you be willing to check out #2211 and see if it addresses the issue?

This is a much larger conversation about fundamental changes, but hoping to at least keep a BC fix going so that projects are uninterrupted as possible until that can happen.

**Comment by AkenRoberts** [comment]
[Released in v3.0.4](https://github.com/doctrine-extensions/DoctrineExtensions/releases/tag/v3.0.4)
